sdk/search/Azure.Search/src/Options/GetDocumentOptions.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+
+namespace Azure.Search
+{
+    /// <summary>
+    /// Options for <see cref="SearchIndexClient.GetDocumentAsync"/>.
+    /// </summary>
+    public class GetDocumentOptions : SearchRequestOptions
+    {
+        /// <summary>
+        /// A list of field names to retrieve.  Only fields marked as
+        /// retrievable can be included in this clause.  Any field not
+        /// retrieved will be missing from the returned document.  If empty,
+        /// all fields marked as retrievable in the schema are returned.
+        /// </summary>
+        public IList<string> SelectedFields { get; set; } = new List<string>();
```

Comment:
Collection properties must not be settable (guidelines). They can be cleared, which an implementation may treat as null. That also simplifies your expression below on line 27 and eliminates the possibility of boxing `0` if `SelectedFields` is null.

Reply:
There's been some discussion around this one.  I'm going to hide the setters for the first preview and we can add them back if needed because that'll be less painful for any earlier adopters than the other direction.  This required wrapping ALL the generator collection properties.